rust/datafusion/src/execution/context.rs

```diff
@@ -1403,6 +1403,128 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn aggregate_timestamps_sum() -> Result<()> {
+        let tmp_dir = TempDir::new()?;
+        let mut ctx = create_ctx(&tmp_dir, 1)?;
+        ctx.register_table("t", test::table_with_timestamps())
+            .unwrap();
+
+        let results = plan_and_collect(
+            &mut ctx,
+            "SELECT sum(nanos), sum(micros), sum(millis), sum(secs) FROM t",
+        )
+        .await
+        .unwrap();
+
+        let expected = vec![
```

Comment:
What are the exact use cases for summing timestamps? When does it make sense?
It looks like PostgreSQL doesn't support it: https://www.postgresql.org/docs/13/functions-aggregate.html

Reply:
That's a good question.  I can think of great uses for avg, and difference (latency between two timestamps), but not really for sum, other than that sum is an important part of computing average?